### PR TITLE
Add unit tests for `Bundler::SSLCerts::CertificateManager`

### PR DESCRIPTION
**Pull request by RochesterinNYC**

### CODE REVIEW

File: spec/bundler/ssl_certs/certificate_manager_spec.rb [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+require "bundler/ssl_certs/certificate_manager"
+
+describe Bundler::SSLCerts::CertificateManager do
+  let(:rubygems_path) { root }
+
+  subject { described_class.new(rubygems_path) }
+
+  # Pretend bundler root is rubygems root
+  before do
+    `mkdir -p #{root}/lib/rubygems/ssl_certs`
```
Comment (segiddins) [inline spec/bundler/ssl_certs/certificate_manager_spec.rb]: Please use ruby methods (fileutils etc)

File: spec/bundler/ssl_certs/certificate_manager_spec.rb [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+  after { `rm -rf #{root}/lib/rubygems` }
+
+  describe "#update_from" do
+    let(:cert_manager) { double(:cert_manager) }
+
+    before { allow(described_class).to receive(:new).with(rubygems_path).and_return(cert_manager) }
+
+    it "should update the certs through a new certificate manager" do
+      allow(cert_manager).to receive(:update!)
+      expect(described_class.update_from!(rubygems_path)).to eq(nil)
```
Comment (segiddins) [inline spec/bundler/ssl_certs/certificate_manager_spec.rb]: be_nil

File: spec/bundler/ssl_certs/certificate_manager_spec.rb [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+    context "when rubygems_path is not nil" do
+      it "should set rubygems_certs" do
+        expect(subject.rubygems_certs).to include(File.join(root, "lib/rubygems/ssl_certs/ssl-cert.pem"))
+      end
+    end
+  end
+
+  describe "#up_to_date?" do
+    let(:rubygems_path) { root }
```
Comment (segiddins) [inline spec/bundler/ssl_certs/certificate_manager_spec.rb]: Redundant

File: spec/bundler/ssl_certs/certificate_manager_spec.rb [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+    context "when bundler certs and rubygems certs are the same" do
+      before do
+        `rm #{root}/lib/rubygems/ssl_certs/ssl-cert.pem`
+        `cp #{root}/lib/bundler/ssl_certs/*.pem #{root}/lib/rubygems/ssl_certs`
+      end
+
+      it "should return true" do
+        expect(subject.up_to_date?).to be_truthy
```
Comment (segiddins) [inline spec/bundler/ssl_certs/certificate_manager_spec.rb]: `expect(subject).to be_up_to_date`?

File: spec/bundler/ssl_certs/certificate_manager_spec.rb [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+      it "should return nil" do
+        expect(subject.update!).to eq(nil)
+      end
+    end
+
+    context "when certificate manager is up to date" do
+      before { allow(subject).to receive(:up_to_date?).and_return(true) }
+
+      it "should return nil" do
+        expect(subject.update!).to eq(nil)
```
Comment (segiddins) [inline spec/bundler/ssl_certs/certificate_manager_spec.rb]: Be_nil

**Comment by RochesterinNYC** [comment]
Updated the PR.

**Comment by segiddins** [comment]
@homu r+

**Comment by RochesterinNYC** [comment]
Thanks!

**Pull request merged**